lib/MusicBrainz/Server/Controller/CDTOC.pm

```diff
@@ -253,10 +254,23 @@ sub attach : Local DenyWhenReadonly
             $c->detach;
         }
         elsif ($search_release->submitted_and_valid($c->req->query_params)) {
+            my $query = $search_release->field('query')->value;
             my $releases = $self->_load_paged($c, sub {
-                $c->model('Search')->search('release', $search_release->field('query')->value, shift, shift,
+                if (is_guid($query)) {
+                    my $release = $c->model('Release')->get_by_gid($query);
+                    return [] unless defined $release;
```

Comment:
I would prefer returning something that specifies that it understood it's an MBID, but it didn't match anything - rather than the standard "No results found. Try refining your search query." it could be something like "We didn't find a release with that MBID" or "That MBID doesn't seem to belong to a release". Not sure what the best way of achieving that would be - I guess you could set a boolean for $was_mbid_search and pass it to the stash?

Reply:
Was hoping I could reuse the message from root/mbid/NotFound.js, but seems a bit too long and generic. Opted for "We couldn’t find a release matching that MBID." Let me know if that can be improved.